apps/api/src/persistence/repositories/CategoryRepository.e2e.spec.ts

```diff
@@ -0,0 +1,126 @@
+import { INestApplication } from '@nestjs/common';
+import createTestModule from '../../app/controllers/__tests__/createTestModule';
+import { InternalError } from '../../lib/errors/InternalError';
+import { NotFound } from '../../lib/types/not-found';
+import buildTestData from '../../test-data/buildTestData';
+import { ArangoConnectionProvider } from '../database/arango-connection.provider';
+import { DatabaseProvider } from '../database/database.provider';
+import { categoryCollectionID } from '../database/types/ArangoCollectionId';
+import { DatabaseDTO } from '../database/utilities/mapEntityDTOToDatabaseDTO';
+import generateRandomTestDatabaseName from './__tests__/generateRandomTestDatabaseName';
+import TestRepositoryProvider from './__tests__/TestRepositoryProvider';
+
+describe('Repository provider > getCategoryRepository', () => {
+    const testDatabaseName = generateRandomTestDatabaseName();
+
+    const testData = buildTestData();
+
+    let arangoConnectionProvider: ArangoConnectionProvider;
+
+    let databaseProvider: DatabaseProvider;
+
+    let testRepositoryProvider: TestRepositoryProvider;
+
+    let app: INestApplication;
+
+    const { categoryTree } = testData;
+
+    beforeAll(async () => {
+        jest.resetModules();
```

Comment:
At this point, it feels like we could abstract the shared test setup for query integration tests.

Reply:
[Here](https://www.pivotaltracker.com/story/show/182199567) is a story for that tech-debt.